common/src/main/java/se/mulander/cosmos/common/model/movies/movie/MovieDaoImpl.java

```diff
@@ -0,0 +1,81 @@
+package se.mulander.cosmos.common.model.movies.movie;
+
+import se.mulander.cosmos.common.database.jpa.Database;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MovieDaoImpl implements se.mulander.cosmos.common.model.movies.movie.MovieDao {
+
+    @Override
+    public List<Movie> getAllMovies() {
+        List<Movie> result = new ArrayList<>();
+        try {
+            result = Database.getObjects("from Movie");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return result;
+    }
+
+    @Override
+    public List<Movie> getMoviesByStatus(String status) {
+        List<Movie> result = new ArrayList<>();
+        Map<String, Object> param = new HashMap<>();
+        param.put("status", status);
+        try {
+            result = Database.getObjects("from Movie WHERE extendedMovie.status = :status", param);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    @Override
+    public Movie getMovieById(String id) {
+        List<Movie> result = new ArrayList<>();
+        Map<String, Object> param = new HashMap<>();
+        param.put("id", id);
+        try {
+            result = Database.getObjects("from Movie WHERE internalID = :id", param);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return result.get(0);
```

Comment:
Kan resultera i en ArrayIndexOutOfBoundsException.
Skulle också föredra om detta returnerade en optional

Reply:
Hmm, kan det verkligen bli det? I fail to see it